src/nodes.js

```diff
@@ -70,11 +71,39 @@ var createElement = function(doc, tag, key, statics) {
  * @return {!Node}
  */
 var createNode = function(doc, nodeName, key, statics) {
+  var node;
   if (nodeName === '#text') {
-    return doc.createTextNode('');
+    node = doc.createTextNode('');
+  } else {
+    node = createElement(doc, nodeName, key, statics);
   }
+  notifications.createdNode(node);
+  return node;
+};
 
-  return createElement(doc, nodeName, key, statics);
+/**
+ * Replaces a child node in the DOM
+ * @param {!Node} parent The child node's parent.
+ * @param {!Node} replacement The replacement node
+ * @param {!Node} child The child node to be removed from the DOM.
+ * @return {!Node}
+ */
+var replaceNode = function(parent, replacement, child) {
```

Comment:
Replacing shouldn't trigger a removal notification, since it may be reinserted later.